src/app/payment/paymentMethod/HostedWidgetPaymentMethod.tsx

```diff
@@ -264,6 +250,40 @@ class HostedWidgetPaymentMethod extends Component<
         );
     }
 
+    renderContainer(shouldShowCreditCardFieldset: any): ReactNode | undefined {
+        const {
+            containerId,
+            hideContentWhenSignedOut = false,
+            hideWidget,
+            isSignInRequired = false,
+            isSignedIn,
+            method,
+            additionalContainerClassName,
+            shouldRenderCustomInstrument = false,
+            validateCustomRender,
+        } = this.props;
+
+        const customContainer = shouldRenderCustomInstrument && validateCustomRender && validateCustomRender();
+
+        return (
+            <div
+                className={ classNames(
+                    'widget',
+                    `widget--${method.id}`,
+                    'payment-widget',
+                    shouldRenderCustomInstrument  ? '' : additionalContainerClassName
+                ) }
+                id={ containerId }
+                style={ {
+                    display: (hideContentWhenSignedOut && isSignInRequired && !isSignedIn) || !shouldShowCreditCardFieldset || hideWidget ? 'none' : undefined,
```

Comment:
instead of hiding it with styles, do you think it's possible not to render this component at all? e.g:
```js
if ((hideContentWhenSignedOut && isSignInRequired && !isSignedIn)
  || !shouldShowCreditCardFieldset || hideWidget
) {
  return null;
}

return (
  <div // ...

Reply:
> instead of hiding it with styles, do you think it's possible not to render this component at all? e.g:
> 
> ```js
> if ((hideContentWhenSignedOut && isSignInRequired && !isSignedIn)
>   || !shouldShowCreditCardFieldset || hideWidget
> ) {
>   return null;
> }
> 
> return (
>   <div // ...
> ```

We must always render the UI element otherwise payment strategy will fail trying to mount the component(s). This is because when vaulted instruments are available we display `CardInstrumentFieldset` or `AccountInstrumentFieldSet` and when shopper opt to enter a different payment instrument we're simply hiding either one and showing the payment form.